### PR TITLE
Feature/edit comment

### PR DESCRIPTION
**Pull request by davidporosnicu**

### CODE REVIEW

**Review by horiaradu — CHANGES_REQUESTED** [review]
This can't be reviewed because it has changes from #9 and #10 :(

**Comment by horiaradu** [comment]
I've temporarily changed the base so that I can review it.

File: src/components/CommentForm.js [hunk 1]
```diff
@@ -65,6 +65,30 @@ export class CommentForm extends Component {
     }
   };
 
+  onUpdateCommentClick = () => {
+    const { author, text, commentDate } = this.state;
+
+    const updatedComment = {
+      author: author
```
Comment (horiaradu) [inline src/components/CommentForm.js]: This code is starting to appear all over the place in the application. In a future PR, please create a utility function which performs this operation and reuse it everywhere.

File: src/redux/actions/blogActions.js [hunk 2]
```diff
@@ -49,3 +49,12 @@ export const deleteComment = (commentUuid, entryUuid2) => async dispatch => {
     entryUuid2
   });
 };
+
+export const updateComment = (commentUuid, updatedComment, entryUuid3) => async dispatch => {
```
Comment (horiaradu) [inline src/redux/actions/blogActions.js]: what is this 3?

File: src/components/Comments.js [hunk 3]
```diff
@@ -1,21 +1,36 @@
...
+import { deleteComment, updateComment } from '../redux/actions/blogActions';
 import { connect } from 'react-redux';
 import '../css/comments.css';
 
 class Comments extends Component {
   state = {
-    showComments: false
+    showComments: false,
+    isEditCommentModeOn: false,
+    currentCommentId: ''
```
Comment (horiaradu) [inline src/components/Comments.js]: Rename flag to `commentIdInEditMode`. However, the presence of this flag is actually showing the wrong abstraction level. You need a component which contains the CommentForm and the edit, delete, save, etc. buttons

File: src/components/Comments.js [hunk 4]
```diff
@@ -1,21 +1,36 @@
...
     this.setState({ showComments: !this.state.showComments });
   };
-
   onDeleteClick = (commentUuid, entryUuid, userId) => {
     this.props.deleteComment(commentUuid, entryUuid, userId);
   };
 
+  onEditCommentClick = (e, commentUuid) => {
+    if (e.target.id === commentUuid) {
+      this.setState({ isEditCommentModeOn: true, currentCommentId: e.target.id, hideEditAndDeleteButtons: true });
```
Comment (horiaradu) [inline src/components/Comments.js]: This flag `hideEditAndDeleteButtons` is useless I think. If `isEditCommentModeOn` is true, the buttons should be hidden.

File: src/components/Comments.js [hunk 5]
```diff
@@ -35,31 +50,57 @@ class Comments extends Component {
...
+                    comment.userId === this.props.currentUser.userId &&
+                    comment.uuid === this.state.currentCommentId ? (
+                      <div>
+                        <EditComment
+                          currentUser={this.props.currentUser}
+                          entryUuid={this.props.entryUuid}
+                          currentComment={comment}
+                          onCancelClick={this.onCancelClick}
+                          onUpdateClick={this.onUpdateClick}
+                          currentCommentId={this.state.currentCommentId}
```
Comment (horiaradu) [inline src/components/Comments.js]: EditComment does not need to receive `currentCommentId` and `currentComment`.

File: src/components/CommentForm.js [hunk 6]
```diff
@@ -110,8 +134,15 @@ export class CommentForm extends Component {
               Enter text
             </p>
           )}
-          <input type="submit" className="commentButton" value="Add Comment" />
+
+          {!this.props.currentComment && <input type="submit" className="commentButton" value="Add Comment" />}
```
Comment (horiaradu) [inline src/components/CommentForm.js]: `currentComment` should not be here. This form should always have these buttons.

File: src/components/EditComment.js [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+const EditComment = props => {
+  return (
+    <div>
+      <CommentForm
+        onCancelClick={props.onCancelClick}
+        currentUser={props.currentUser}
+        entryUuid={props.entryUuid}
+        currentComment={props.currentComment}
+        onUpdateClick={props.onUpdateClick}
+        currentCommentId={props.currentCommentId}
```
Comment (horiaradu) [inline src/components/EditComment.js]: This should not receive `currentCommentId`.

**Review by horiaradu — CHANGES_REQUESTED** [review]

File: src/components/CommentForm.js [hunk 8]
```diff
@@ -110,8 +134,15 @@ export class CommentForm extends Component {
...
         </form>
+        {this.props.currentComment && (
```
Comment (horiaradu) [inline src/components/CommentForm.js]: Put these buttons inside the `EditComment` component

File: src/components/CommentForm.js (hunk 6 above)
Comment (davidporosnicu) [reply src/components/CommentForm.js]: if i delete that condition whenever i want to edit a comment the 'Add Comment' button will be there too

Comment (horiaradu) [reply src/components/CommentForm.js]: why don't you use your newly created method: `isEditModeOn()`?

File: src/services/api.js [hunk 9]
```diff
@@ -126,6 +126,39 @@ const api = {
...
     return filteredComments;
+  },
+  async updateComment(commentUuid, updatedComment, entryUuid) {
+    const snapshot = await firebase
+      .firestore()
+      .collection('entries')
+      .get();
+
+    let commentsArray = [];
+    snapshot.docs.map(x => {
```
Comment (horiaradu) [inline src/services/api.js:137]: What is the use of these lines? You are doing a `map`, but you don't use the result. Also, you are not doing any mutations in the `map` (which is good). But, leads me to believe that the code is useless.

**Review by horiaradu — APPROVED** [review]

**Review by horiaradu — APPROVED** [review]

**Pull request merged**